### PR TITLE
Feat/brush event

### LINKED ISSUE #164: [Feature] Rich Brush event, emit such `brushEnd` event
### What problem does this feature solve?

support `brushEnd` event

### What does the proposed API look like?

```ts
cs.on('brushEnd', e => {
  // we can get what selected in params
})
```

### PR DESCRIPTION
**Pull request by skie1997**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #164 #165 

### 💡 Background and solution
- brush support 'brushStart' | 'brushChange' | 'brushEnd' event
- add element datum in params 
20230714114321_rec_

```ts
this.event.emit(eventType, {
          model: this,
          value: {
            // 操作类型
            operateType,
            // 正在操作的region
            operateRegion: region,
            // 在选框内的 element data
            inBrushData: this._extendDataInBrush(this._inBrushElementsMap),
            // 在选框外的 element data
            outOfBrushData: this._extendDatumOutOfBrush(this._outOfBrushElementsMap),
            // 被链接的系列中：在选框内的 element data
            linkInBrushData: this._extendDataInBrush(this._linkedInBrushElementsMap),
            // 被链接的系列中：在选框外的 element data
            linkOutOfBrushData: this._extendDatumOutOfBrush(this._linkedOutOfBrushElementsMap),
            // 在选框内的 vgrammar elements
            inBrushElementsMap: this._inBrushElementsMap,
            // 在选框外的 vgrammar elements
            outOfBrushElementsMap: this._outOfBrushElementsMap,
            // 被链接的系列中：在选框内的 vgrammar elements
            linkedInBrushElementsMap: this._linkedInBrushElementsMap,
            // 被链接的系列中：在选框外的 vgrammar elements
            linkedOutOfBrushElementsMap: this._linkedOutOfBrushElementsMap
          }
        });
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | rish brush event of operateType and datum |
| 🇨🇳 Chinese | 丰富brush组件透出的事件类型并在参数中增加元素数据 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: common/changes/@visactor/lark-vchart/feat-brush-event_2023-07-14-03-19.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@visactor/lark-vchart",
+      "comment": "feat(brush): add operate type and export data",
```
Comment (kkxxkk2019) [inline common/changes/@visactor/lark-vchart/feat-brush-event_2023-07-14-03-19.json:5]: 注释信息再完善一下，怎么了什么事件，同时在强调是在事件参数上到处了什么数据

**Comment by kkxxkk2019** [comment]
记得网站的 API 文档也更新一下

**Comment by skie1997** [comment]
> 记得网站的 API 文档也更新一下

已更新

**Review by kkxxkk2019 — APPROVED** [review]

**Comment by skie1997** [comment]
merge

**Pull request merged**